components/eamxx/src/physics/mam/eamxx_mam_microphysics_process_interface.cpp

```diff
@@ -267,249 +514,377 @@ void MAMMicrophysics::initialize_impl(const RunType run_type) {
-
-      // do aerosol microphysics (gas-aerosol exchange, nucleation, coagulation)
-      impl::modal_aero_amicphys_intr(config.amicphys, step_, dt, t, pmid, pdel,
-                                     zm, pblh, qv, cldfrac, vmr, vmrcw, vmr_pregaschem,
-                                     vmr_precldchem, vmrcw_precldchem, vmr_tendbb,
-                                     vmrcw_tendbb, dgncur_a, dgncur_awet,
-                                     wetdens, qaerwat);
-
-      //-----------------
-      // LINOZ chemistry
-      //-----------------
-
-      // the following things are diagnostics, which we're not
-      // including in the first rev
-      Real do3_linoz, do3_linoz_psc, ss_o3, o3col_du_diag, o3clim_linoz_diag,
-           zenith_angle_degrees;
-
-      // FIXME: Need to get chlorine loading data from file
-      Real chlorine_loading = 0.0;
-
-      Real rlats = col_lat * M_PI / 180.0; // convert column latitude to radians
-      int o3_ndx = 0; // index of "O3" in solsym array (in EAM)
-      mam4::lin_strat_chem::lin_strat_chem_solve_kk(o3_col_dens_i(k), temp,
-        zenith_angle, pmid, dt, rlats,
-        linoz_o3_clim(icol, k), linoz_t_clim(icol, k), linoz_o3col_clim(icol, k),
-        linoz_PmL_clim(icol, k), linoz_dPmL_dO3(icol, k), linoz_dPmL_dT(icol, k),
-        linoz_dPmL_dO3col(icol, k), linoz_cariolle_psc(icol, k),
-        chlorine_loading, config.linoz.psc_T, vmr[o3_ndx],
-        do3_linoz, do3_linoz_psc, ss_o3,
-        o3col_du_diag, o3clim_linoz_diag, zenith_angle_degrees);
-
-      // update source terms above the ozone decay threshold
-      if (k > nlev - config.linoz.o3_lbl - 1) {
-        Real do3_mass; // diagnostic, not needed
-        mam4::lin_strat_chem::lin_strat_sfcsink_kk(dt, pdel, vmr[o3_ndx], config.linoz.o3_sfc,
-          config.linoz.o3_tau, do3_mass);
-      }
+#if defined(ENABLE_OUTPUT_TRACER_FIELDS)
+  const auto ver_emiss_fields_outputs =
+      get_field_out("vertical_emission_fields").get_view<Real ***>();
+  for(int ifield = 0; ifield < int(ver_emiss_fields_outputs.extent(1));
+      ++ifield) {
+    const auto field_at_i = Kokkos::subview(
+        ver_emiss_fields_outputs, Kokkos::ALL(), ifield, Kokkos::ALL());
+    Kokkos::deep_copy(field_at_i, vert_emis_output_[ifield]);
+  }
+#endif
 
-      // ... check for negative values and reset to zero
-      for (int i = 0; i < gas_pcnst; ++i) {
-        if (vmr[i] < 0.0) vmr[i] = 0.0;
-      }
+  const_view_1d &col_latitudes     = col_latitudes_;
+  const_view_1d &d_sfc_alb_dir_vis = d_sfc_alb_dir_vis_;
 
-      //----------------------
-      // Dry deposition (gas)
-      //----------------------
+  mam_coupling::DryAtmosphere &dry_atm = dry_atm_;
+  mam_coupling::AerosolState &dry_aero = dry_aero_;
 
-      // FIXME: C++ port in progress!
-      //mam4::drydep::drydep_xactive(...);
+  mam4::mo_photo::PhotoTableData &photo_table = photo_table_;
+  const Config &config                        = config_;
```

Comment:
We are creating a soft copy of `config_`, i.e., `config`, because we cannot use a private member of a class within parallel_for. This approach allows us to bypass the issue. @bartgol, is there another way to fix this issue? Note: While this issue initially appears as a warning, it results in a runtime error when executed on NVIDIA GPUs.